### PR TITLE
QML Whitelisting + Commerce Integration

### PR DESCRIPTION
**Pull request by zfox23**
Big changes here that impact Commerce and the Tablet/HUD. Huge thanks to @jherico for doing _all_ of the QML whitelisting work in this PR.

To QA this PR, please perform the following tests (I know this is a lot, but this PR greatly affects the tablet and toolbar):
https://highfidelity.testrail.net/index.php?/cases/view/2185
https://highfidelity.testrail.net/index.php?/cases/view/2188
https://highfidelity.testrail.net/index.php?/cases/view/89
https://highfidelity.testrail.net/index.php?/cases/view/93
https://highfidelity.testrail.net/index.php?/cases/view/115

Thanks to you angelic QA testers.

### CODE REVIEW

**Comment by Nex-Pro** [comment]
Awaiting win build

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4950/HighFidelity-Beta-PR11815-8f855c6c9568dc921783914818a859fcea3baeff.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4950/HighFidelity-Beta-PR11815-8f855c6c9568dc921783914818a859fcea3baeff.dmg

**Comment by Nex-Pro** [comment]
Testing

**Comment by misslivirose** [comment]
@zfox23 -- how will this impact marketplace tablet apps that exist already? When will the Tablet version of the API be removed?

**Comment by Nex-Pro** [comment]
@zfox23 You mentioned the 'Angelic' QA testers. But you are out of luck, I am testing it Muhaaaa!

**Comment by zfox23** [comment]
@misslivirose This _shouldn't_ impact existing tablet apps at all. This PR is only supposed to add new functionality. If it breaks anything with existing apps, that's a bug.

I don't understand your question about "the Tablet version of the API" - could you please clarify?

@Nex-Pro UH OH.

**Comment by misslivirose** [comment]
@zfox23 Right now it looks like both Tablets and Tablet point to the same code with a TODO to remove the Tablet global object

**Comment by zfox23** [comment]
Aha, thanks for clarifying. I didn't put that code in there, so I'm not sure what Austin had in mind. @jherico, can you please comment?

**Comment by Nex-Pro** [comment]
### Fail on first two Testrail cases. Filed.

**Comment by Nex-Pro** [comment]
https://highfidelity.testrail.net/index.php?/tests/view/13328&group_by=cases:section_id&group_order=asc&group_id=440

https://highfidelity.testrail.net/index.php?/tests/view/13338&group_by=cases:section_id&group_order=asc&group_id=441

**Comment by zfox23** [comment]
Thanks, @Nex-Pro. @jherico has made an update. Could you please begin re-testing?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4964/HighFidelity-Beta-PR11815-0ad399974132417c0181d35af6dd46806730022e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4964/HighFidelity-Beta-PR11815-0ad399974132417c0181d35af6dd46806730022e.dmg

**Comment by Nex-Pro** [comment]
Will do. @zfox23

**Comment by Nex-Pro** [comment]
@zfox23 Fails in both testrail cases.
https://highfidelity.testrail.net/index.php?/tests/view/13328&group_by=cases:section_id&group_order=asc&group_id=440

https://highfidelity.testrail.net/index.php?/tests/view/13338&group_by=cases:section_id&group_order=asc&group_id=441

**Comment by zfox23** [comment]
@Nex-Pro could you please provide logs for those failure cases?

**Comment by Nex-Pro** [comment]
Logs submitted via slack @zfox23

**Comment by zfox23** [comment]
Some relevant errors from the logs:
```
[11/15 22:32:50] [WARNING] [default] QQmlContext: Cannot set property on internal context.
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:196: ReferenceError: Commerce is not defined
[11/15 22:32:50] [DEBUG] [qml] [WalletHome.qml] User scrolled to the bottom of 'Recent Activity'.
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseModal.qml:261: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/SecurityImageChange.qml:73: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:196: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Help.qml:30:5: QML Connections: Cannot assign to non-existent property "onKeyFilePathIfExistsResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Security.qml:30:5: QML Connections: Cannot assign to non-existent property "onKeyFilePathIfExistsResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletHome.qml:31:5: QML Connections: Cannot assign to non-existent property "onHistoryResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletHome.qml:31:5: QML Connections: Cannot assign to non-existent property "onBalanceResult"
[11/15 22:32:50] [DEBUG] [qml] [WalletHome.qml] User scrolled to the bottom of 'Recent Activity'.
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseModal.qml:39:5: QML Connections: Cannot assign to non-existent property "onWalletAuthenticatedStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseModal.qml:39:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/SecurityImageChange.qml:33:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:39:5: QML Connections: Cannot assign to non-existent property "onChangePassphraseStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:39:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:39:5: QML Connections: Cannot assign to non-existent property "onChangePassphraseStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:39:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletSetup.qml:40:5: QML Connections: Cannot assign to non-existent property "onKeyFilePathIfExistsResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletSetup.qml:40:5: QML Connections: Cannot assign to non-existent property "onWalletAuthenticatedStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletSetup.qml:40:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Wallet.qml:41:5: QML Connections: Cannot assign to non-existent property "onSecurityImageResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Wallet.qml:41:5: QML Connections: Cannot assign to non-existent property "onLoginStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Wallet.qml:41:5: QML Connections: Cannot assign to non-existent property "onWalletStatusResult"
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Wallet.qml:259: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Wallet.qml:42: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletSetup.qml:41: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:40: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseSelection.qml:40: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/SecurityImageChange.qml:34: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/NeedsLogIn.qml:34: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/PassphraseModal.qml:40: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/WalletHome.qml:32: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/SendMoney.qml:29: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Security.qml:31: ReferenceError: Commerce is not defined
[11/15 22:32:50] [WARNING] [default] file:///E:/Program Files/High Fidelity - PR11815/resources/qml/hifi/commerce/wallet/Help.qml:31: ReferenceError: Commerce is not defined
```

```
[11/15 22:30:07] [DEBUG] [hifi.scriptengine] stopTimer -- not in _timerFunctionMap QObject(0x0)
[11/15 22:30:07] [CRITICAL] [hifi.scriptengine] [defaultScripts.js] [UncaughtException signalHandlerException] Error: Result of expression 'button' [undefined] is not an object. in file:///E:/Program Files/High Fidelity - PR11815/scripts/system/commerce/wallet.js:163
[11/15 22:30:07] [CRITICAL] [hifi.scriptengine] [Backtrace]
[11/15 22:30:07] [CRITICAL] [hifi.scriptengine]     shutdown() at file:///E:/Program Files/High Fidelity - PR11815/scripts/system/commerce/wallet.js:163
[11/15 22:30:07] [CRITICAL] [hifi.scriptengine]     <global>() at -1
[11/15 22:30:07] [DEBUG] [hifi.controllers] Attempting to  disable  mapping  "file:///E:/Program Files/High Fidelity - PR11815/scripts/system/libraries/entitySelectionTool.js-click"
[11/15 22:30:07] [WARNING] [default] Unable to find MenuUserData for object  QAction(0x13026ed9290 text="" menuRole=TextHeuristicRole visible=false)
[11/15 22:30:07] [WARNING] [default] ""
[11/15 22:30:07] [WARNING] [default] Attempted to remove menu action with no found QML object
```

```

[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
[11/15 22:31:42] [WARNING] [default] qrc:///qml/hifi/tablet/TabletHome.qml:124: ReferenceError: wrapper is not defined
```

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4992/HighFidelity-Beta-PR11815-93c3d5314421256547d8518885993f91e76387b1.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4992/HighFidelity-Beta-PR11815-93c3d5314421256547d8518885993f91e76387b1.dmg

**Comment by Nex-Pro** [comment]
@zfox23 Hit me up in slack when this is ready.

**Comment by zfox23** [comment]
@Nex-Pro Ready!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5062/HighFidelity-Beta-PR11815-d31578abae3a1310b8224e6505a19866c1c54ebd.dmg

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5079/HighFidelity-Beta-PR11815-d31578abae3a1310b8224e6505a19866c1c54ebd.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5079/HighFidelity-Beta-PR11815-d31578abae3a1310b8224e6505a19866c1c54ebd.dmg

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
need HFC/wallet @zfox23

**Comment by Nex-Pro** [comment]
FAIL on tablet basic interactions https://highfidelity.testrail.net/index.php?/tests/view/14489&group_by=cases:section_id&group_order=asc&group_id=27

**Comment by Nex-Pro** [comment]
failed on finger and stylus tablet interactions.
https://highfidelity.testrail.net/index.php?/runs/view/1034&group_by=cases:section_id&group_order=asc

**Comment by Nex-Pro** [comment]
@zfox23 I forgot to add a note that I saw the toolbar and tablet were open at the same time.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5474/HighFidelity-Beta-PR11815-b2bafed2bb074433a6053092016a942b9aeabe39.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5474/HighFidelity-Beta-PR11815-b2bafed2bb074433a6053092016a942b9aeabe39.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5524/HighFidelity-Beta-PR11815-a81a981676e8400ea9c6281f419f9881ccb881a4.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5524/HighFidelity-Beta-PR11815-a81a981676e8400ea9c6281f419f9881ccb881a4.dmg

**Comment by Nex-Pro** [comment]
Testing by request shortly.

**Comment by Nex-Pro** [comment]
In progress

File: libraries/ui/src/QmlWindowClass.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ QVariantMap QmlWindowClass::parseArguments(QScriptContext* context) {
 
     QUrl url { properties[SOURCE_PROPERTY].toString() };
     if (url.scheme() != "http" && url.scheme() != "https" && url.scheme() != "file" && url.scheme() != "about" &&
```
Comment (NissimHadar) [inline libraries/ui/src/QmlWindowClass.cpp]: A comment here would be helpful, as you are already changing the code.

Comment (zfox23) [reply libraries/ui/src/QmlWindowClass.cpp]: @jherico, any insight here?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5571/HighFidelity-Beta-PR11815-b81cdf49ef5fecce8eace6ebe6f122cf6967313a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5571/HighFidelity-Beta-PR11815-b81cdf49ef5fecce8eace6ebe6f122cf6967313a.exe

File: libraries/ui/src/QmlWindowClass.cpp (hunk 1 above)
Comment (zfox23) [reply libraries/ui/src/QmlWindowClass.cpp]: (Resolved on Slack, thanks Austin)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5573/HighFidelity-Beta-PR11815-2d786451c2a8be3221b9f4c6c618a0b0a4ca10b8.exe

**Comment by howard-stearns** [comment]
looks like macOS build was aborted for no reason

**Comment by howard-stearns** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5574/HighFidelity-Beta-PR11815-2d786451c2a8be3221b9f4c6c618a0b0a4ca10b8.dmg

**Comment by howard-stearns** [comment]
oh come on jenkins

**Comment by howard-stearns** [comment]
build this please

**Comment by zfox23** [comment]
😡

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5576/HighFidelity-Beta-PR11815-2d786451c2a8be3221b9f4c6c618a0b0a4ca10b8.exe

**Comment by marko8904** [comment]
this is just awful. Either a Mac fails or this PR encounters a weird Windows problem. build this please.

**Comment by zfox23** [comment]
😭

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5578/HighFidelity-Beta-PR11815-2d786451c2a8be3221b9f4c6c618a0b0a4ca10b8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5578/HighFidelity-Beta-PR11815-2d786451c2a8be3221b9f4c6c618a0b0a4ca10b8.dmg

**Comment by Nex-Pro** [comment]
Fail Test 1 As discussed with Zach
Fail Test 2 As discussed with Zach


Test 3 blocked. Avatars do not persist long enough to carry out a reliable test at dev-welcome.  ( known issue, currently being investigated )
https://highfidelity.testrail.net/index.php?/tests/view/23952&group_by=cases:section_id&group_order=asc&group_id=25

Test 4 FLickable is broken.

https://highfidelity.testrail.net/index.php?/tests/view/24019&group_by=cases:section_id&group_order=asc&group_id=26

Is working fine in #7456

Test 5 Unable to reasonably proceed with https://highfidelity.testrail.net/index.php?/runs/view/1528&group_by=cases:section_id&group_id=27&group_order=asc

As the cases are for toolbar interactions but are titled for tablet.

**Comment by zfox23** [comment]
Thanks @Nex-Pro .

**Comment by jherico** [comment]
@Nex-Pro do you mean working find in 7546 ?  7456 is ancient.

**Comment by zfox23** [comment]
@Nex-Pro I fixed the Wallet Setup bug.

Please see @jherico's comment about the flickable bug so that we can get this back to QA Ready post-haste.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5586/HighFidelity-Beta-PR11815-2225352b1e151d574d03356838a3e1e10e65f703.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5586/HighFidelity-Beta-PR11815-2225352b1e151d574d03356838a3e1e10e65f703.dmg

**Comment by Nex-Pro** [comment]
@jherico 7546 was latest on dev download earlier today

**Comment by zfox23** [comment]
@Nex-Pro - I think @jherico understands that, but you wrote "7456" in your original comment.

You used "7546" for the baseline test, right?

**Comment by Nex-Pro** [comment]
@zfox23 Yeah just a typo

**Comment by Nex-Pro** [comment]
I told you it was 7546 in slack, You could have passed that on

**Comment by zfox23** [comment]
@Nex-Pro I had no reason to remember that what you told me was different from what you put here, and I didn't go back and check. But you're right, I could have.

**Comment by zfox23** [comment]
So - @jherico - that means there's a bug in this PR, right?

**Comment by changxc** [comment]
Hi all 

just so that we are on the same page - 

Test 4 FLickable is broken.

https://highfidelity.testrail.net/index.php?/tests/view/24019&group_by=cases:section_id&group_order=asc&group_id=26

do we have any video / picture / logs or anything to help developer debugging? 

Ideally we wanted this PR to be in RC 61, since it address a security concern. Thank you so much!


@Nex-Pro @jherico @zfox23 @ZappoMan @howard-stearns @divyaelango88

**Comment by Nex-Pro** [comment]
### FAIL

### Test 3 is now a **PASS** as the avatars persisted and allowed testing of the people list in desktop/toolbar mode as required by the specified test rail case.
https://highfidelity.testrail.net/index.php?/tests/view/24141&group_by=cases:section_id&group_order=asc&group_id=25

Proceeding with next test.
### Test 3 Tablet basic interactions with styus and finger Flickable failed.
I have made videos of the flickable fail and of the behaviour in 7550 where flickable works. I also tried 7554 but had no lasers and could only interact with the tablet using the mouse. Flickable worked with the mouse in the market with 7554. I find it odd that I had to do this. Let me know where you want me to send the vids. IMO too large to dump here.

To recap the flickable/scrolling with inertia problem. **It does not work in 11815.**
 It works in 7550 and 7554 and 7546. So that is a comparison to three other builds. I hope this is thorough enough.

This section is a pass apart from the Flickable/scrolling/iniertia not working in the market or on the create app. 

https://highfidelity.testrail.net/index.php?/tests/view/24019&group_by=cases:section_id&group_order=asc&group_id=26 The testrail case itself is in need of triage and I have posted on fogbugz.
https://highfidelity.fogbugz.com/f/cases/10528/Testrail-case-T24019-needs-reworking.

### Test 5. Tested against toolbar as otherwise most test steps would be incorrect. Looks like the tests are mis-labeled.

3 failed sub sections. Flicking fails.
https://highfidelity.testrail.net/index.php?/runs/view/1530&group_by=cases:section_id&group_id=27&group_order=asc

in addition saw new bug on QML keyboard.

If the collapse button is clicked  the keyboard can remain active and can be typed on. However the text field is not in focus. Does not happen in latest dev build.

11815_keyboard_issue2
11815_keyboard_issue

**Comment by Nex-Pro** [comment]
@changxc  Here are the videos for the flick scrolling failures and comparisons to 7550 ( and 7554 )

http://s3-us-west-1.amazonaws.com/hifi-makers/pioneer-nexial.protagonist/QA_Testing/11815_Flicking_Not_Working.flv

Videos showing flicking working in 7550
http://s3-us-west-1.amazonaws.com/hifi-makers/pioneer-nexial.protagonist/QA_Testing/7550_flicking_working.flv
http://s3-us-west-1.amazonaws.com/hifi-makers/pioneer-nexial.protagonist/QA_Testing/7550_flicking_working_in_create_properties.flv

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5617/HighFidelity-Beta-PR11815-376444527271c82f359f6d1a04b200250b83f2c3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5617/HighFidelity-Beta-PR11815-376444527271c82f359f6d1a04b200250b83f2c3.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5709/app_11815_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5710/app_11815_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5710/HighFidelity-Beta-PR11815-afd3cf7045e8a452b243cb7742655f0286107e76.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5710/HighFidelity-Beta-PR11815-afd3cf7045e8a452b243cb7742655f0286107e76.dmg

**Comment by divyaelango88** [comment]
hey guys, I want to say, I installed this PR on my computer every time I try to to shut down the sandbox, it does not get shutdown. It says shutting down for the longest time and it never goes away. 
capture

it might not be related to this pr but it happens consistently on it

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5721/app_11815_PR.apk

**Comment by zfox23** [comment]
@divyaelango88 I cannot reproduce this problem, thankfully!

**Comment by divyaelango88** [comment]
@zfox23 yeah I deleted my comment, I don't think it had anything to do with this pr, but thanks for checking :)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5721/HighFidelity-Beta-PR11815-18d5be147c36f05bdafcceefb2d21e7f7d0be94b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5721/HighFidelity-Beta-PR11815-18d5be147c36f05bdafcceefb2d21e7f7d0be94b.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
[Updating as I go along]

For the Wallet based test runs (the first 2):
https://highfidelity.testrail.net/index.php?/runs/view/1642&group_by=cases:section_id&group_order=asc
**50% Pass and complete**
For the Tablet & Toolbar based test runs (the last 3): https://highfidelity.testrail.net/index.php?/runs/view/1643&group_by=cases:section_id&group_order=asc

**0%**

**Comment by YuppyPlays** [comment]
Testing complete:

**For the Wallet based test runs (the first 2):**
https://highfidelity.testrail.net/index.php?/runs/view/1642&group_by=cases:section_id&group_order=asc
**For the Tablet & Toolbar based test runs (the last 3):** https://highfidelity.testrail.net/index.php?/runs/view/1643&group_by=cases:section_id&group_order=asc

Notes: Just had to restart the 2nd test as it turns out my Steam somehow crashed (the Steam dashboard) and this had an effect on the game which stopped the "Login screen" on tablet to function, it just showed a "Sign in with Steam" button which did nothing- and a Cancel button. Upon rebooting my Vive and the SteamVR software, this was back to normal and so far grants most if not all of the 2 tests as Passes
More Notes: Found a strange bug which is not in current master 7586, using tablet and clicking Menu > File > Login doesn't seem to work, brings up a blank page of black/dark gray background with "File" at the top only.

I would say this is a **Pass** but the issue I noted could make this a fail, I did not experience the mentioned issue about the "Menu > File > Login" not working in Master (7586).

**Comment by zfox23** [comment]
Will investigate the possible fail you mentioned, THANK YOU!

**Comment by zfox23** [comment]
Can't repro the bug mentioned; discussing on Slack.

**Pull request merged**